h2o-core/src/main/java/water/init/NetworkInit.java

```diff
@@ -394,7 +395,11 @@ public static void initializeNetworkSockets( ) {
 
     // Late instantiation of Jetty object, if needed.
     if (H2O.getJetty() == null && !H2O.ARGS.disable_web) {
-      H2O.setJetty(new JettyHTTPD());
+      if(H2O.ARGS.disable_web) {
+        H2O.setJetty(new DummyServer());
```

Comment:
I did not get this part, why do you need to setup DummyServer when disable_web is specified?